cloudify_agent/api/factory.py

```diff
@@ -196,18 +196,18 @@ def save(self, daemon):
             self.storage, '{0}.json'.format(
                 daemon.name)
         )
+        cluster_path = os.path.join(
+            self.storage, 'cluster-{0}.json'.format(
+                daemon.name)
+        )
+        daemon.cluster_settings_path = cluster_path
```

Comment:
set that path always - even if we don't have a cluster yet, we can join it later without restarting the agent - and by then, we'll need to have already decided on the path